frontends/ol-utilities/src/learning-resources/learning-resources.ts

```diff
@@ -116,12 +120,59 @@ const formatRunDate = (
   return null
 }
 
+/**
+ * Checks if all runs of a given learning resource are identical in terms of price, delivery method, and location.
+ *
+ * @param resource - The learning resource to check.
+ * @returns `true` if all runs have the same price, delivery method, and location; otherwise, `false`.
+ */
+const allRunsAreIdentical = (resource: LearningResource) => {
+  if (!resource.runs) {
+    return true
+  }
+  if (resource.runs.length <= 1) {
+    return true
+  }
+  const prices: LearningResourcePrice[] = []
+  const deliveryMethods = []
+  const locations = []
+  for (const run of resource.runs) {
+    if (run.resource_prices) {
+      run.resource_prices.forEach((price) => {
+        if (price.amount !== "0") {
+          prices.push(price)
+        }
+      })
+    }
+    if (run.delivery) {
+      deliveryMethods.push(run.delivery)
+    }
+    if (run.location) {
+      locations.push(run.location)
+    }
+  }
+  const distinctPrices = [...new Set(prices.map((p) => p.amount).flat())]
```

Comment:
These work fine, but shorter to use Set throughout instead of converting between arrays, ie.

```
136:    const prices = new Set<LearningResourcePrice>()

144:    run.resource_prices?.forEach(price => {
            if (price.amount !== "0") prices.add(price.amount);
        });

161:    prices.size === 1